backend/src/chat/messageStore.js

```diff
@@ -0,0 +1,60 @@
+const mongoose = require('mongoose');
+const Message = require('./Message');
+
+const conversationSchema = new mongoose.Schema({
+    users: { type: [String], required: true },
+    messages: { type: [Object], default: [] },
+});
+
+const Conversation = mongoose.model('Conversation', conversationSchema);
+
+class MessageStore {
+    constructor() {
+        if (!MessageStore.instance) {
+            MessageStore.instance = this;
+        }
+        return MessageStore.instance;
+    }
```

Comment:
Is this an Singleton?

Reply:
Yes, there was originally some initialization logic in the constructor to make it less computationally intensive when you create a message store, but I realized you don't need it. So now we just create one and never allocate space for another one.

Reply:
Understandable. Actually, I think the FE currently is not using the Http client as a singletone and might be an idea worth exploring for them for the future. Not sure if we need to initilize an http client each time we are about to send a new request unless we resort to some other way of using the http client.